resolwe/flow/executors/startup_communication_container.py

```diff
@@ -572,12 +572,10 @@ async def start(self) -> int:
             logger.debug("Terminating upload thread.")
             upload_thread.terminate()
             upload_thread.join()
-            if not KEEP_DATA:
-                purge_secrets()
-
-            if not await self.collect_produced_files():
-                if return_code == 0:
-                    return_code = 1
+            # Secrets are read-only in kubernetes.
+            if not RUNNING_IN_KUBERNETES:
+                if not KEEP_DATA:
+                    purge_secrets()
```

Comment:
Will this work when the system will try to create the same secret for the second time?

Reply:
Secrets are actually kubernetes objects and are created/purged by the kubernetes workload connector/dispatcher. They can not be deleted.